### PR TITLE
Add C# SDK v5.x snippet for ip-messaging

### PR DESCRIPTION
**Pull request by well1791**

### CODE REVIEW

File: ip-messaging/users/token-generation-server/token-generation-server.5.x.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Linq;
```
Comment (acamino) [inline ip-messaging/users/token-generation-server/token-generation-server.5.x.cs]: Are you using `System.Linq`? Is usually a good practice to do not include imports that won't be used in this file. You can use your editor to remove the unused imports.

File: ip-messaging/users/token-generation-server/token-generation-server.5.x.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            var identity = Internet.UserName();
+
+            // Create an IP messaging grant for this token
+            var grant = new IpMessagingGrant();
+            grant.EndpointId = $"TwilioChatDemo:{identity}:{device}";
+            grant.ServiceSid = ipmServiceSid;
+
+            var grants = new HashSet<IGrant>
+            {
+              { grant }
```
Comment (acamino) [inline ip-messaging/users/token-generation-server/token-generation-server.5.x.cs]: Although the extra curly braces are not an error, they introduced an additional scope. Would you mind removing them?

File: ip-messaging/users/token-generation-server/token-generation-server.5.x.cs (hunk 1 above)
Comment (well1791) [inline ip-messaging/users/token-generation-server/token-generation-server.5.x.cs]: @acamino not sure it works on atom.. 🤔

File: ip-messaging/users/token-generation-server/token-generation-server.5.x.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            var accountSid = ConfigurationManager.AppSettings["TwilioAccountSid"];
+            var apiKey = ConfigurationManager.AppSettings["TwilioApiKey"];
+            var apiSecret = ConfigurationManager.AppSettings["TwilioApiSecret"];
+            var ipmServiceSid = ConfigurationManager.AppSettings["TwilioIpmServiceSid"];
+
+            // Create a random identity for the client
+            var identity = Internet.UserName();
+
+            // Create an IP messaging grant for this token
+            var grant = new IpMessagingGrant();
```
Comment (acamino) [inline ip-messaging/users/token-generation-server/token-generation-server.5.x.cs]: Did you consider on using C# object literal notation?

File: ip-messaging/users/token-generation-server/token-generation-server.5.x.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            var accountSid = ConfigurationManager.AppSettings["TwilioAccountSid"];
+            var apiKey = ConfigurationManager.AppSettings["TwilioApiKey"];
+            var apiSecret = ConfigurationManager.AppSettings["TwilioApiSecret"];
+            var ipmServiceSid = ConfigurationManager.AppSettings["TwilioIpmServiceSid"];
+
+            // Create a random identity for the client
+            var identity = Internet.UserName();
+
+            // Create an IP messaging grant for this token
+            var grant = new IpMessagingGrant
```
Comment (acamino) [inline ip-messaging/users/token-generation-server/token-generation-server.5.x.cs]: What about removing this `grant` variable? Probably use something like:

```csharp
var grants = new HashSet<IGrant>
{
    new IpMessagingGrant
    {
        EndpointId = $"TwilioChatDemo:{identity}:{device}",
        ServiceSid = ipmServiceSid
    }
};
```

I would extract different variables if this example will require more than one grant to increase the snippet readability.

Comment (well1791) [inline ip-messaging/users/token-generation-server/token-generation-server.5.x.cs]: actually that's a great suggestion! thanks for pointing it!

**Comment by acamino** [comment]
Thanks @well1791! LGTM

**Pull request merged**